src/main/java/org/elasticsearch/rest/action/admin/indices/analyze/RestAnalyzeAction.java

```diff
@@ -64,4 +73,58 @@ public void handleRequest(final RestRequest request, final RestChannel channel,
         analyzeRequest.charFilters(request.paramAsStringArray("char_filters", analyzeRequest.charFilters()));
         client.admin().indices().analyze(analyzeRequest, new RestToXContentListener<AnalyzeResponse>(channel));
     }
+
+    public static void parseContent(AnalyzeRequest request, String text, BytesReference content) {
```

Comment:
Maybe the the parsing logic should be in `TransportAnalyzeAction` instead? This way both rest and regular transport level requests deal with the same parameter parsing logic.

The AnalyzeRequest just holds a `source` field , like search and percolate request. To make usage easier from transport client, there can be a AnalyzeSourceBuilder (similar to SearchSourceBuilder).